core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2ServiceModule.java

```diff
@@ -57,7 +57,9 @@ protected void setup(Binder binder)
     @ForOAuth2
     public static SigningKeyResolver createSigningKeyResolver(OAuth2Config oauth2Config, @ForOAuth2 HttpClient httpClient)
     {
-        return new JwkSigningKeyResolver(new JwkService(URI.create(oauth2Config.getJwksUrl()), httpClient, new Duration(15, TimeUnit.MINUTES)));
+        JwkService jwkService = new JwkService(URI.create(oauth2Config.getJwksUrl()), httpClient, new Duration(15, TimeUnit.MINUTES));
+        jwkService.start();
```

Comment:
This method is annotated with `@PostConstruct` it should be already called, however it is not called because we do not bind `JwkService` directly.

I think we should bind `JwkService` in guice directly. However it requires some refactoring of how `JwkService` is created with `@Inject` constructor as here you have different config class.

Maybe a follow up?